src/wallet/rpcwallet.cpp

```diff
@@ -3031,9 +3033,13 @@ UniValue listunspent(const JSONRPCRequest& request)
         if (fValidAddress) {
             entry.pushKV("address", EncodeDestination(address));
 
-            if (pwallet->mapAddressBook.count(address)) {
-                entry.pushKV("label", pwallet->mapAddressBook[address].name);
-                entry.pushKV("account", pwallet->mapAddressBook[address].name);
+            {
+                LOCK(pwallet->cs_wallet);
```

Comment:
@laanwj how about locking the wallet for the whole loop?

Reply:
Depends on the relative overhead of locking/unlocking for every look-up here. If that is negligible, it's preferable to lock/unlock with the small granularity like you have now.

Reply:
This PR aims to reduce `cs_main` and that is already accomplished.

Keeping the wallet locked for the whole loop is no worse than the previous implementation. Overhead aside, this implementation is more susceptible to an unspent change, so I would say to lock the whole loop.

Reply:
> Keeping the wallet locked for the whole loop 

@laanwj changed to this.